src/ngraph/pass/nop_elimination.cpp

```diff
@@ -134,62 +134,263 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node)
     return false;
 }
 
+static size_t count_unknown_dims(const PartialShape& ps)
+{
+    size_t rc = 0;
+    if (ps.is_static())
+    {
+        return rc;
+    }
+    for (auto i = 0; i < ps.rank().get_length(); i++)
+    {
+        if (ps[i].is_dynamic())
+        {
+            rc += 1;
+        }
+    }
+    return rc;
+}
+
+static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node)
+{
+    auto shape_ps = node->get_output_partial_shape(0);
+    if (shape_ps.rank().get_length() == 0)
+    {
+        return false;
+    }
+    if (count_unknown_dims(shape_ps) > 1)
+    {
+        return false;
+    }
+    std::vector<int64_t> target_shape;
```

Comment:
Why not to use ngraph::Shape here?

Reply:
this is 2nd argument to reshape, it expects a node (target_shape is used to create it)